.github/workflows/rust.yml

```diff
@@ -22,10 +22,27 @@ jobs:
       with:
         command: bootimage
         args: --target=x86_64-mycelium.json
-    - name: Test
+
+  test:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+    - name: install rust toolchain
+      uses: actions-rs/toolchain@v1
+      with:
+        profile: minimal
+        toolchain: nightly
+        components: rust-src, llvm-tools-preview
+    - name: install dev env
+      uses: actions-rs/cargo@v1.0.1
+      with:
+        command: dev-env
+    - name: install qemu
+      run: sudo apt-get update && sudo apt-get install qemu
+    - name: run tests
```

Comment:
would be good to have a step that runs any "regular" (i.e. not in qemu) rust tests in crates as well. i think we define some now as of #64, so it would be good for them to run on CI as well. 

if this was a separate job, it could run in parallel with the qemu tests.

Reply:
Yeah, we should probably do that. We'll probably want to make sure to exclude the root `mycelium_kernel` crate, though.

Reply:
we might have to just `cargo test -p $FOO` for every other crate...which is fine, it's just a little flaky since it needs to be manually updated as we add more crates...

Reply:
I ended up just using more `#[cfg(...)]` attributes to stub out a dummy `fn main()` for the `mycelium_kernel` crate when we're running on the host platform.